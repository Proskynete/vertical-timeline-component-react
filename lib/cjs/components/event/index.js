"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Event = void 0;
var react_1 = __importStar(require("react"));
var useConfig_1 = require("../../hooks/useConfig");
var main_1 = require("../../styles/main");
var Event = function (_a) {
    var _b;
    var defaultClosed = _a.defaultClosed, description = _a.description, title = _a.title;
    var config = (0, useConfig_1.useConfig)().config;
    var _c = (0, react_1.useState)(!title ? true : !defaultClosed), show = _c[0], setShow = _c[1];
    var isCollapsed = config.collapse;
    var handleSetShow = function () {
        setShow(!show);
    };
    return (react_1.default.createElement(react_1.default.Fragment, null,
        !!title && (react_1.default.createElement(main_1.EventTitle, { collapse: isCollapsed, onClick: function () {
                if (isCollapsed)
                    handleSetShow();
            }, style: (_b = config.customStyles) === null || _b === void 0 ? void 0 : _b.event },
            isCollapsed && react_1.default.createElement(main_1.Icon, { isShowing: show }, "\u203A"),
            title)),
        show && (react_1.default.createElement(main_1.DescriptionWrapper, null, description.map(function (text, i) {
            var _a;
            return (react_1.default.createElement(main_1.Description, { key: i, symbol: config.descriptionSymbol, style: (_a = config.customStyles) === null || _a === void 0 ? void 0 : _a.description }, text));
        })))));
};
exports.Event = Event;
