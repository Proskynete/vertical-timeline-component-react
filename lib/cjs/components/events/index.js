"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Events = void 0;
var react_1 = __importDefault(require("react"));
var useConfig_1 = require("../../hooks/useConfig");
var main_1 = require("../../styles/main");
var event_1 = require("../event");
var validate_1 = require("../validate");
var year_content_1 = __importDefault(require("../year-content"));
var Events = function (_a) {
    var _b;
    var title = _a.title, startDate = _a.startDate, endDate = _a.endDate, active = _a.active, withoutDay = _a.withoutDay, children = _a.children;
    var config = (0, useConfig_1.useConfig)().config;
    return (react_1.default.createElement(main_1.ContainerWrapper, null,
        react_1.default.createElement(year_content_1.default, { startDate: startDate, endDate: endDate, active: active, withoutDay: withoutDay }),
        react_1.default.createElement(main_1.BodyWrapper, null,
            react_1.default.createElement(main_1.Title, { style: (_b = config.customStyles) === null || _b === void 0 ? void 0 : _b.title }, title),
            react_1.default.createElement(main_1.BodyInner, null,
                react_1.default.createElement(validate_1.Validate, { componentToValidate: event_1.Event }, children)))));
};
exports.Events = Events;
