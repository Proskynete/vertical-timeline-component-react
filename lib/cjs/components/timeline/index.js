"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Timeline = void 0;
var react_1 = __importDefault(require("react"));
var styled_components_1 = require("styled-components");
var config_1 = require("../../config");
var config_context_1 = require("../../context/config.context");
var main_1 = require("../../styles/main");
var events_1 = require("../events");
var validate_1 = require("../validate");
var Timeline = function (_a) {
    var children = _a.children, _b = _a.collapse, collapse = _b === void 0 ? config_1.defaultValues.collapse : _b, customStyles = _a.customStyles, _c = _a.dateFormat, dateFormat = _c === void 0 ? config_1.defaultValues.dateFormat : _c, _d = _a.descriptionSymbol, descriptionSymbol = _d === void 0 ? config_1.defaultValues.descriptionSymbol : _d, _e = _a.lang, lang = _e === void 0 ? config_1.defaultValues.lang : _e, _f = _a.theme, theme = _f === void 0 ? config_1.defaultValues.theme : _f, _g = _a.withoutDay, withoutDay = _g === void 0 ? config_1.defaultValues.withoutDay : _g;
    return (react_1.default.createElement(main_1.TimelineWrapper, null,
        react_1.default.createElement(main_1.TimelineWrapperInner, null,
            react_1.default.createElement(config_context_1.ConfigProvider, { config: {
                    theme: theme,
                    lang: lang,
                    dateFormat: dateFormat,
                    collapse: collapse,
                    customStyles: customStyles,
                    descriptionSymbol: descriptionSymbol,
                    withoutDay: withoutDay,
                } },
                react_1.default.createElement(styled_components_1.ThemeProvider, { theme: theme },
                    react_1.default.createElement(validate_1.Validate, { componentToValidate: events_1.Events }, children))))));
};
exports.Timeline = Timeline;
