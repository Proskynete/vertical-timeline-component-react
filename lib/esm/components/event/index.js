import React, { useState } from 'react';
import { useConfig } from '../../hooks/useConfig';
import { Description, DescriptionWrapper, EventTitle, Icon } from '../../styles/main';
var Event = function (_a) {
    var _b;
    var defaultClosed = _a.defaultClosed, description = _a.description, title = _a.title;
    var config = useConfig().config;
    var _c = useState(!title ? true : !defaultClosed), show = _c[0], setShow = _c[1];
    var isCollapsed = config.collapse;
    var handleSetShow = function () {
        setShow(!show);
    };
    return (React.createElement(React.Fragment, null,
        !!title && (React.createElement(EventTitle, { collapse: isCollapsed, onClick: function () {
                if (isCollapsed)
                    handleSetShow();
            }, style: (_b = config.customStyles) === null || _b === void 0 ? void 0 : _b.event },
            isCollapsed && React.createElement(Icon, { isShowing: show }, "\u203A"),
            title)),
        show && (React.createElement(DescriptionWrapper, null, description.map(function (text, i) {
            var _a;
            return (React.createElement(Description, { key: i, symbol: config.descriptionSymbol, style: (_a = config.customStyles) === null || _a === void 0 ? void 0 : _a.description }, text));
        })))));
};
export { Event };
