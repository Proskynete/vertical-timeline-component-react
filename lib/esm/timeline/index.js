import React from 'react';
import { shape, node, oneOf, oneOfType, arrayOf } from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { defaultValues } from '../../config';
import { ConfigContext } from '../../context/config.context';
import { TimelineWrapper, TimelineWrapperInner } from '../../styles/main';
var Timeline = function (props) {
    var theme = props.theme, lang = props.lang, children = props.children, dateFormat = props.dateFormat;
    return (React.createElement(TimelineWrapper, null,
        React.createElement(TimelineWrapperInner, null,
            React.createElement(ConfigContext.Provider, { value: { lang: lang, dateFormat: dateFormat } },
                React.createElement(ThemeProvider, { theme: theme }, children)))));
};
Timeline.defaultProps = {
    theme: {
        yearColor: defaultValues.theme.yearColor,
        lineColor: defaultValues.theme.lineColor,
        dotColor: defaultValues.theme.dotColor,
        borderDotColor: defaultValues.theme.borderDotColor,
        titleColor: defaultValues.theme.titleColor,
        subtitleColor: defaultValues.theme.subtitleColor,
        textColor: defaultValues.theme.textColor,
    },
    lang: defaultValues.lang,
    dateFormat: defaultValues.dateFormat,
};
Timeline.propTypes = {
    theme: shape({}),
    lang: oneOf(['es', 'en', 'de', 'tr']),
    dateFormat: oneOf(['L', 'l', 'll']),
    children: oneOfType([arrayOf(node), node]).isRequired,
};
export default Timeline;
