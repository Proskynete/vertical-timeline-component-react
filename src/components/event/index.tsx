import React, { useState } from 'react';
import { useConfig } from '../../hooks/useConfig';
import { ContentProps } from '../../interfaces';
import { Description, DescriptionWrapper, EventTitle, Icon } from '../../styles/main';

const Event = ({ title, description, collapse, defaultClosed }: ContentProps) => {
	const { config } = useConfig();
	const [show, setShow] = useState(!defaultClosed);
	const isCollapsed = collapse || config.collapse;

	const handleSetShow = () => {
		setShow(!show);
	};

	return (
		<>
			<EventTitle
				collapse={isCollapsed}
				onClick={() => {
					if (isCollapsed) handleSetShow();
				}}
				style={config.customStyles?.event}
			>
				{isCollapsed && <Icon isShowing={show}>&#8250;</Icon>}
				{title}
			</EventTitle>

			{show && (
				<DescriptionWrapper>
					{description.map((text, i) => (
						<Description
							key={i}
							symbol={config.descriptionSymbol}
							style={config.customStyles?.description}
						>
							{text}
						</Description>
					))}
				</DescriptionWrapper>
			)}
		</>
	);
};

export { Event };
