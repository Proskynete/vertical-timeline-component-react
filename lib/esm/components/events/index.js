import React from 'react';
import { useConfig } from '../../hooks/useConfig';
import { ContainerWrapper, BodyWrapper, Title, Subtitle, BodyInner } from '../../styles/main';
import { Event } from '../event';
import { Validate } from '../validate';
import YearContent from '../year-content';
var Events = function (_a) {
    var _b, _c;
    var active = _a.active, children = _a.children, _d = _a.defaultClosed, defaultClosed = _d === void 0 ? false : _d, endDate = _a.endDate, startDate = _a.startDate, subtitle = _a.subtitle, title = _a.title, withoutDay = _a.withoutDay;
    var config = useConfig().config;
    return (React.createElement(ContainerWrapper, null,
        React.createElement(YearContent, { startDate: startDate, endDate: endDate, active: active, withoutDay: withoutDay }),
        React.createElement(BodyWrapper, null,
            React.createElement(Title, { style: (_b = config.customStyles) === null || _b === void 0 ? void 0 : _b.title }, title),
            subtitle && React.createElement(Subtitle, { style: (_c = config.customStyles) === null || _c === void 0 ? void 0 : _c.subtitle }, subtitle),
            React.createElement(BodyInner, null,
                React.createElement(Validate, { componentToValidate: Event, defaultClosed: defaultClosed }, children)))));
};
export { Events };
