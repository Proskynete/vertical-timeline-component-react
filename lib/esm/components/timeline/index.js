import React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultValues } from '../../config';
import { ConfigProvider } from '../../context/config.context';
import { TimelineWrapper, TimelineWrapperInner } from '../../styles/main';
import { Events } from '../events';
import { Validate } from '../validate';
var Timeline = function (_a) {
    var children = _a.children, _b = _a.collapse, collapse = _b === void 0 ? defaultValues.collapse : _b, customStyles = _a.customStyles, _c = _a.dateFormat, dateFormat = _c === void 0 ? defaultValues.dateFormat : _c, _d = _a.descriptionSymbol, descriptionSymbol = _d === void 0 ? defaultValues.descriptionSymbol : _d, _e = _a.lang, lang = _e === void 0 ? defaultValues.lang : _e, _f = _a.theme, theme = _f === void 0 ? defaultValues.theme : _f, _g = _a.withoutDay, withoutDay = _g === void 0 ? defaultValues.withoutDay : _g;
    return (React.createElement(TimelineWrapper, null,
        React.createElement(TimelineWrapperInner, null,
            React.createElement(ConfigProvider, { config: {
                    theme: theme,
                    lang: lang,
                    dateFormat: dateFormat,
                    collapse: collapse,
                    customStyles: customStyles,
                    descriptionSymbol: descriptionSymbol,
                    withoutDay: withoutDay,
                } },
                React.createElement(ThemeProvider, { theme: theme },
                    React.createElement(Validate, { componentToValidate: Events }, children))))));
};
export { Timeline };
