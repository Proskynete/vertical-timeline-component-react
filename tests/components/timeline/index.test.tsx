import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import { Timeline } from '../../../src/components/timeline';
import { Events } from '../../../src/components/events';
import { Event } from '../../../src/components/event';

describe('Timeline', () => {
	test('should create a snapshot of timeline component', async () => {
		const component = render(
			<Timeline>
				<Events title="I'm a title" startDate="2019/01/01" today>
					<Event title="I'm a event title" description={["I'm a description"]} />
				</Events>
			</Timeline>,
		);

		expect(component).toMatchSnapshot();
	});

	test('should render only one container but with endDate prop', () => {
		render(
			<Timeline collapse>
				<Events title="I'm a title" startDate="2019/01/01" endDate="2020/01/01">
					<Event title="I'm a event title" description={["I'm a description"]} />
				</Events>
			</Timeline>,
		);
	});

	test('should render timeline with all custom props', async () => {
		const theme = {
			yearColor: '#000000',
			lineColor: '#000000',
			dotColor: '#000000',
			borderDotColor: '#000000',
			titleColor: '#000000',
			subtitleColor: '#000000',
			textColor: '#000000',
		};

		render(
			<Timeline lang="es" theme={theme} dateFormat="with-weekday" collapse withoutDay>
				<Events title="I'm a title" startDate="2019/01/01" endDate="2020/01/01">
					<Event title="I'm a event title" description={["I'm a description"]} />
				</Events>
			</Timeline>,
		);
	});

	test('should hide description when user has click on title event', async () => {
		render(
			<Timeline collapse>
				<Events title="I'm a title" startDate="2019/01/01" today>
					<Event title="I'm a event title" description={["I'm a description"]} />
				</Events>
			</Timeline>,
		);

		const title = screen.getByText("I'm a title");
		const eventTitle = screen.getByText("I'm a event title");
		const description = screen.getByText("I'm a description");

		expect(title).toBeInTheDocument();
		expect(eventTitle).toBeInTheDocument();
		expect(description).toBeInTheDocument();

		await userEvent.click(eventTitle);
		expect(eventTitle).toHaveStyle('cursor: pointer');

		await userEvent.click(eventTitle);
		expect(description).not.toBeInTheDocument();
	});

	test('should change styles with customStyles props', async () => {
		render(
			<Timeline
				customStyles={{
					title: { color: 'red' },
					event: { color: 'red' },
					date: { color: 'red' },
					description: { color: 'red' },
				}}
			>
				<Events title="I'm a title" startDate="2019/01/01" today>
					<Event title="I'm a event title" description={["I'm a description"]} />
				</Events>
			</Timeline>,
		);

		const title = screen.getByText("I'm a title");
		const eventTitle = screen.getByText("I'm a event title");
		const description = screen.getByText("I'm a description");

		expect(title).toHaveStyle('color: red');
		expect(eventTitle).toHaveStyle('color: red');
		expect(description).toHaveStyle('color: red');
	});
});
