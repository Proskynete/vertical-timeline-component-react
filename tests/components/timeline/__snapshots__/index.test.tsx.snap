// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should create a snapshot of timeline component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ cyRdvt"
      >
        <div
          class="sc-gsnTZi kEhIuA"
        >
          <section
            class="sc-dkzDqf bqgjsq"
          >
            <div
              aria-label="From Tuesday, January 01, 2019 To Monday, December 26, 2022"
              class="sc-hKMtZM VwYTb"
              format="only-number"
              lang="en"
            >
              <time
                aria-hidden="true"
                datetime="12/26/2022"
              >
                2022
              </time>
              <span
                aria-hidden="true"
                class="sc-eCYdqJ ctJLYX"
              >
                From
              </span>
              <time
                aria-hidden="true"
                datetime="1/1/2019"
              >
                1/1/2019
              </time>
            </div>
            <article
              class="sc-jSMfEi htGDEx"
            >
              <h2
                class="sc-gKXOVf dFwsvn"
              >
                I'm a title
              </h2>
              <p
                class="sc-iBkjds icZsBo"
              >
                I'm a subtitle
              </p>
              <div
                class="sc-ftvSup ccSJqk"
              >
                <h3
                  class="sc-jqUVSM gpKZtt"
                >
                  I'm a event title
                </h3>
                <div
                  class="sc-kDDrLX bFQhiG"
                >
                  <p
                    class="sc-iqcoie bIyfuL"
                  >
                    I'm a description
                  </p>
                </div>
              </div>
            </article>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ cyRdvt"
    >
      <div
        class="sc-gsnTZi kEhIuA"
      >
        <section
          class="sc-dkzDqf bqgjsq"
        >
          <div
            aria-label="From Tuesday, January 01, 2019 To Monday, December 26, 2022"
            class="sc-hKMtZM VwYTb"
            format="only-number"
            lang="en"
          >
            <time
              aria-hidden="true"
              datetime="12/26/2022"
            >
              2022
            </time>
            <span
              aria-hidden="true"
              class="sc-eCYdqJ ctJLYX"
            >
              From
            </span>
            <time
              aria-hidden="true"
              datetime="1/1/2019"
            >
              1/1/2019
            </time>
          </div>
          <article
            class="sc-jSMfEi htGDEx"
          >
            <h2
              class="sc-gKXOVf dFwsvn"
            >
              I'm a title
            </h2>
            <p
              class="sc-iBkjds icZsBo"
            >
              I'm a subtitle
            </p>
            <div
              class="sc-ftvSup ccSJqk"
            >
              <h3
                class="sc-jqUVSM gpKZtt"
              >
                I'm a event title
              </h3>
              <div
                class="sc-kDDrLX bFQhiG"
              >
                <p
                  class="sc-iqcoie bIyfuL"
                >
                  I'm a description
                </p>
              </div>
            </div>
          </article>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
